lookout/style/format/quality_report.py

```diff
@@ -36,19 +71,304 @@ def quality_report(input_pattern: str, bblfsh: str, language: str, n_files: int,
+                    self.log.warning("Failed to parse %s", file.path)
+                    continue
+                X, y, vnodes_y, vnodes = res
+                y_pred, rule_winners = rules.predict(X, vnodes_y, vnodes, lang)
+                self.log.debug("y.shape %s" % y.shape)
+                self.log.debug("len(vnodes_y) %s" % len(vnodes_y))
+                self.log.debug("y_pred.shape %s" % y_pred.shape)
+
+                if self.config["aggregate"] is True:
+                    agg_y.append(y)
+                    agg_vnodes_y.extend(vnodes_y)
+                    agg_y_pred.append(y_pred)
+                else:
+                    target_names = [CLASSES[cls_ind] for cls_ind in numpy.unique(y)]
+                    assert len(y) == len(y_pred)
+
+                    report = self.generate_report(
+                        y=y, y_pred=y_pred, vnodes_y=vnodes_y, target_names=target_names,
+                        config=self.config, model=self.model, feature_extractor=fe,
+                        rule_winners=rule_winners
+                    )
+                    comments.append(generate_comment(filename=file.path, confidence=100,
+                                                     line=0, text=report))
+            model_report = self.generate_model_report(self.model, self.config, fe)
+            if model_report:
+                comments.append(generate_comment(filename=file.path, confidence=100,
+                                                 line=0, text=model_report))
+            if self.config["aggregate"]:
+                agg_y = numpy.hstack(agg_y)
+                target_names = [CLASSES[cls_ind] for cls_ind in numpy.unique(agg_y)]
+                report = self.generate_report(
+                    y=agg_y, y_pred=numpy.hstack(agg_y_pred), vnodes_y=agg_vnodes_y,
+                    target_names=target_names, config=self.config, model=self.model,
+                    feature_extractor=fe, rule_winners=rule_winners
+                )
+                comments.append(generate_comment(filename=file.path, confidence=100,
+                                                 line=0, text=report))
+        return comments
+
+    @classmethod
+    @with_uasts_and_contents
+    def train(cls, ptr: ReferencePointer, config: MutableMapping[str, Any],
+              data_request_stub: DataStub, **data) -> None:
+        """
+        Analyze a set of changes from one revision to another and make report for all files.
+
+        :param ptr: Git repository state pointer.
+        :param config: configuration dict.
+        :param data: contains "files" - the list of files in the pointed state.
+        :param data_request_stub: connection to the Lookout data retrieval service, not used.
+        :return: AnalyzerModel containing the learned rules, per language.
+        """
+        config = cls._load_analyze_config(config)
+        assert config["model_path"] is not None, "'model_path' is missed in configuration. " \
```

Comment:
Better to go to a new line after `config["model_path"] is not None`

Reply:
Still holds

Reply:
missed this last comment